packages/training/python/src/training/action_executor.py

```
@@ -0,0 +1,621 @@
+            ticker=ticker,
+            side=direction,
+            size=size,
+            entry_price=perp.mark_price,
+            new_balance=self.portfolio.balance,
+            new_positions=self.portfolio.position_count,
+        )
+    
+    def _execute_perp_close(self, action: Dict) -> ActionResult:
+        """Execute perpetual position close"""
+        ticker = action["ticker"]
+        size = action["size"]
```

Ruff (0.14.10)

Local variable `size` is assigned to but never used

Remove assignment to unused variable `size`

(F841)